### PR TITLE
Feature 1144 model table

### PR DESCRIPTION
**Pull request by MugPand**
# Title of Pull Request HERE

Github Issue Number Here: #1144 
**What user problem are we solving?**
The DynamoDB Table structures were recently overhauled, so the endpoints had to be updated to reflect the changes.
**What solution does this PR provide?**
Added POST, GET, GET ALL, DELETE, DELETE ALL endpoints with unit tests to ensure functionality. 
**Testing Methodology**
> How did you test your changes and verify that existing functionality is not broken
Utilized Postman to validate all endpoints with sample data. Also, created and verified unit tests for each endpoint. Followed the same standard set by issue #1155.
**Any other considerations**

### CODE REVIEW

File: serverless/packages/functions/src/model/create_model.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        const eventBody = JSON.parse(event.body? event.body : "");
+        const putCommand: PutItemCommand = new PutItemCommand({
+            TableName: "ModelTable",
+            Item:
+            {
+                user_id: {"S": user_id},
+                model_id: {"S": model_id},
+                name: {"S": eventBody['name']},
+                model_structure: {"S": eventBody['model_structure']}
```
Comment (karkir0003) [inline serverless/packages/functions/src/model/create_model.ts:21]: did we ever discuss how model_structure would be populated in dynamo?

@noah-iversen @codingwithsurya @alantao912

File: serverless/packages/functions/src/model/delete_model.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import { DynamoDBClient, DeleteItemCommand } from '@aws-sdk/client-dynamodb';
+
+export async function handler(event : APIGatewayProxyEventV2) {
+    let queryParams = null;
+    if (event && (queryParams = event['pathParameters']) != null) {
+        const model_id: string | undefined = queryParams['model_id'];
+        if (model_id == undefined) {
+            return {
+                statusCode: 401,
+                body: JSON.stringify({message: "Malformed request content - model ID missing."})
```
Comment (karkir0003) [inline serverless/packages/functions/src/model/delete_model.ts:11]: does this error message structure exist for other dynamo endpoints for trainspace and user?

File: serverless/packages/functions/src/model/delete_model.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            }
+        });
+
+        const response = await client.send(command);
+
+        if (response.$metadata.httpStatusCode == undefined || response.$metadata.httpStatusCode != 200) 
+        {
+            return {
+                statusCode: 404,
+                body: JSON.stringify({ message : "Delete operation failed" })
```
Comment (karkir0003) [inline serverless/packages/functions/src/model/delete_model.ts:31]: are there any cloudwatch logs that get populated for each call of these API gateway endpoints? might be useful to be able to identify which model id we attempted to delete that resulted in the failure?

**Review by karkir0003 — APPROVED** [review]
LGTM. Added a few questions. please address those along with the Sonarcloud checks

File: serverless/packages/functions/src/model/delete_model.ts (hunk 2 above)
Comment (MugPand) [reply serverless/packages/functions/src/model/delete_model.ts:11]: yup, the error handling is standard across all of the trainspace, user, and model endpoints.

File: serverless/packages/functions/src/model/delete_model.ts (hunk 3 above)
Comment (MugPand) [reply serverless/packages/functions/src/model/delete_model.ts:31]: To my knowledge, cloudwatch logs aren't populated for the API gateway endpoints outside of startup/whenever they get hit/billing info. Not sure if custom logs can be added, but interesting as something to look into.

Comment (karkir0003) [reply serverless/packages/functions/src/model/delete_model.ts:31]: lets make that a followup

**Pull request merged**